rs/nervous_system/agent/src/sns/root.rs

```diff
@@ -28,4 +34,30 @@ impl RootCanister {
             )
             .await
     }
+
+    pub async fn list_sns_canisters<C: CallCanisters>(&self, agent: &C) -> Result<Sns, C::Error> {
+        let ListSnsCanistersResponse {
+            root: Some(sns_root_canister_id_1),
+            governance: Some(sns_governance_canister_id),
+            ledger: Some(sns_ledger_canister_id),
+            swap: Some(swap_canister_id),
+            index: Some(index_canister_id),
+            archives,
+            dapps: _,
+        } = agent
+            .call(self.canister_id, ListSnsCanistersRequest {})
+            .await?
+        else {
+            panic!("SNS root canister did not return canister IDs for all canisters - this should never happen");
```

Comment:
I wonder how long before someone uses this crate inside a canister. 

Would it be better to return an error (instead of panicking) to be prepared for that case?

Reply:
Done!